src/Control/Director.php

```diff
@@ -76,40 +80,47 @@ class Director implements TemplateGlobalProvider
     private static $alternate_base_folder;
 
     /**
+     * Setting this explicitly specifies the protocol ("http" or "https") used, overriding the normal
+     * behaviour of Director::is_https introspecting it from the request
+     *
      * @config
      *
-     * @var array
+     * @var string
      */
-    protected static $dev_servers = array();
+    private static $alternate_protocol;
 
     /**
-     * @config
+     * Setting this overrides the host name
      *
-     * @var array
+     * @config
+     * @var string
      */
-    protected static $test_servers = array();
+    private static $alternate_host;
 
     /**
-     * Setting this explicitly specifies the protocol ("http" or "https") used, overriding the normal
-     * behaviour of Director::is_https introspecting it from the request. False values imply default
-     * introspection.
+     * Force the base_url to a specific value.
+     * If assigned, default_base_url and the value in the $_SERVER
+     * global is ignored.
+     * Supports back-ticked vars; E.g. '`SS_BASE_URL`'
      *
      * @config
-     *
      * @var string
      */
-    private static $alternate_protocol;
+    private static $alternate_base_url;
```

Comment:
Default base URL is only used when the URL can't be inferred from the request, e.g. in a CLI call. This ensures that websites can function on >1 domain.

Alternate base URL is used in all cases and is intended for cases like multi-server static publisher setups and subsites where you want to deliberately override.

Admittedly it could be tidier eg we could have a Boolean "force_base_url" flag that switches to the latter behaviour.